lib/vrf.c

```diff
@@ -873,20 +873,51 @@ void vrf_install_commands(void)
 	install_element(ENABLE_NODE, &no_vrf_debug_cmd);
 }
 
+/* this function is here to append running config
+ * in daemons not using vrfs.
+ */
+static int vrf_config_write_stub(struct vty *vty)
+{
+	return 0;
+}
+
+/* this function is here to append running config
+ * in daemons using vrfs. the vrf will be displayed
+ * in running-config
+ */
+int vrf_config_write_generic(struct vty *vty)
+{
+	struct vrf *vrf;
+
+	RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
```

Comment:
Again this breaks assumptions about how we do not expect to output modes if they have no sub-mode data.

Reply:
even daemons that do not have interface node configurable, will have to have those lines.
the reason is that when you are on vtysh, and you try to access interface node, all daemons will be questioned.

and if at least one daemon answers: not possible, then the command will fail from vtysh perspective.

the only way to tackle this problem is:
- either use daemon config node ( by telnet localhost XXX)
- or permit vrf creation for all vrfs that will be on the configuration of at least one daemon.